superdesk/editor_utils.py

```diff
@@ -0,0 +1,427 @@
+        self._mapping[value.key] = value.data
+
+
+class Block:
+    """Abstraction of DraftJS block"""
+
+    def __init__(self, editor, data=None):
+        if data is None:
+            data = {
+                "key": str(uuid.uuid4()),
+                "text": "",
+                "type": "unstyled",
+                "depth": 0,
+                "inlineStyleRanges": [],
+                "entityRanges": [],
+                "data": {},
+            }
+        self.data = data
+        self.entities = EntitySequence(editor, self)
+
+    @property
+    def depth(self):
+        return self.data['depth']
+
+    @property
+    def type(self):
+        return self.data['type']
+
+    @property
+    def text(self):
+        return self.data.get('text')
+
+    def __str__(self):
+        return "{type} block with text {text!r} and {len_entities} entities".format(
+            type=self.type, text=self.text, len_entities=len(self.entities))
+
+    def __repr__(self):
+        return str(self)
+
+
+class EmbedBlock(Block):
+
+    def __init__(self, editor, html):
+        """
+        :param str html: raw HTML to embed
+        """
+        super().__init__(editor)
+        self.data.update({
+            "text": " ",
+            "type": "atomic",
+        })
+        entity_data = {
+            "type": "EMBED",
+            "mutability": "MUTABLE",
+            "data": {
+                "data": {
+                    "html": html,
+                }
+            },
+        }
+        entity = Entity(
+            ranges={"offset": 0, "length": 1, "key": editor.get_next_entity_key()},
+            data=entity_data,
+        )
+        self.entities.append(entity)
+
+
+class BlockSequence(MutableSequence):
+
+    def __init__(self, editor):
+        self.editor = editor
+        self._blocks = editor.content_state.setdefault('blocks', {})
+
+    def __getitem__(self, key):
+        return Block(self.editor, self._blocks[key])
+
+    def __setitem__(self, key, value):
+        if not isinstance(value, Block):
+            raise TypeError("a Block instance is expected")
+        self._blocks[key] = value.data
+
+    def __delitem__(self, key):
+        del self._blocks[key]
+
+    def __len__(self):
+        return len(self._blocks)
+
+    def insert(self, index, value):
+        if not isinstance(value, Block):
+            raise TypeError("a Block instance is expected")
+        self._blocks.insert(index, value.data)
+
+
+class EditorContent:
+    """Base class to manage content from editors"""
+
+    def __init__(self):
+        raise RuntimeError("This class must not be instantiated directly, use create()")
+
+    @staticmethod
+    def create(item, field):
+        """EditorContent factory"""
+        if 'fields_meta' in item:
+            return Editor3Content(item, field)
+        else:
+            raise NotImplementedError("This is not an editor 3 item, EditorContent doesn't manage this")
+
+
+class Editor3Content(EditorContent):
```

Comment:
I think it would still make sense to split ContentState modifications and rendering, then compose them if needed

Reply:
can be, I'll update this later.

Reply:
done, the HTML exporter is now separated. That make more sense indeed, and it let room to properly implement other exporter in the future if needed (like plain text exporter).